### ISSUE TITLE
Improve the Hugo snap package

### ISSUE THREAD

**Issue by jmooring** [gohugoio/hugo#9078]
## Background

The Hugo snap package does not include external dependencies required for certain Hugo features. Under [strict&nbsp;confinement](https://snapcraft.io/docs/snap-confinement) these dependencies must be bundled within the snap.

We have an [open request](https://forum.snapcraft.io/t/19892) for [classic&nbsp;confinement](https://snapcraft.io/docs/snap-confinement) which would allow us to remove the dependencies from the snap[^1], but this would:

1. Result in site building errors if the dependencies have not been manually installed
2. Force users to manually install the dependencies (a non-trivial task in some cases)

We should try to make strict confinement work as much as possible.

## Dependencies

External Dependency|v0.88.1|v0.101.0
:--|:--|:--
Asciidoctor|:x:|:heavy_check_mark:
Embedded Dart Sass|:x:|:heavy_check_mark:
Git|:heavy_check_mark:|:heavy_check_mark:
Go|:x:|:heavy_check_mark:
Node.js|:x:[^2]|:heavy_check_mark:
Pandoc|:x:[^3]|:heavy_check_mark:
rst2html|:x:|:heavy_check_mark:

[^1]: Removing bundled dependencies is not the primary motivation behind our request for classic confinement.
[^2]: Based on [this comment](https://github.com/gohugoio/hugo/pull/9029#issuecomment-950146281), the Node.js version is too old (v10) on some platforms. We should be bundling v12.20 at a minimum, preferably v16.
[^3]: The existing snapcraft.yaml includes a part for pandoc, but it is not working. "error while loading shared libraries: libatomic.so.1: cannot open shared object file: No such file or directory"

## Testing

1) Install the extended version of Hugo from the Snap Store.

```bash
sudo snap install hugo --channel=extended/stable
```

2) Clone this test repository, install the npm packages, and build the site.

```bash
git clone https://github.com/jmooring/hugo-snap-test
cd hugo-snap-test
npm install
/snap/bin/hugo
```

**Comment by Letme** [comment]
Ping on this, as it seems like long lasting problem that was never fixed. It also means there is no decent installation option for newer versions of hugo (apt has version 0.92)

**Comment by anthonyfok** [comment]
@Letme Thank you for the heads up!  I didn't realize that Ubuntu kinetic (the future 22.10 release) has Hugo stuck on 0.92.2-1 when it is already 0.101.0-1 on Debian.  It seems to be due to one of the updated dependencies (aws-sdk-go) not being synchronized from Debian.  The Ubuntu developers have just been notified and will hopefully have the issue resolved soon so that you'll get Hugo v0.101.0 in Ubuntu 22.10.

Meanwhile, about this issue: It might be partially solved (Node.js and pandoc are _probably_ updated and working), but I'll need to find time to actually sit down, test, and work on any remaining issues.  Sorry for the delay!

**Comment by anthonyfok** [comment]
Dear all,

I believe this has just been fixed for v0.101.0 in the `stable` branch.  Please install/refresh the hugo snap to v0.101.0+git4.d7294751 (13802) in the extended/stable channel.

See #10174 for the pull request for the `master` branch.

Due to the recently improved security model in Hugo, the following patch needs to be applied to https://github.com/jmooring/hugo-snap-test for the test to pass:

```diff
diff --git a/config/_default/config.toml b/config/_default/config.toml
index 3bf5e3a..6306b75 100644
--- a/config/_default/config.toml
+++ b/config/_default/config.toml
@@ -15,3 +15,7 @@ target = 'content'
 
 [params]
 transpiler = 'dartsass'  # dartsass or libsass
+
+[security]
+  [security.exec]
+    allow = ['^asciidoctor$', '^dart-sass-embedded$', '^go$', '^npx$', '^pandoc$', '^postcss$', '^rst2html$']
```
